ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/controllerCore/command/ControllerCoreCommandDataHolder.java

```diff
@@ -0,0 +1,203 @@
+package us.ihmc.commonWalkingControlModules.controllerCore.command;
+
+import us.ihmc.commonWalkingControlModules.controllerCore.WholeBodyControllerCoreMode;
+import us.ihmc.commonWalkingControlModules.controllerCore.command.feedbackController.FeedbackControlCommandList;
+import us.ihmc.commonWalkingControlModules.controllerCore.command.inverseDynamics.InverseDynamicsCommandList;
+import us.ihmc.commonWalkingControlModules.controllerCore.command.inverseKinematics.InverseKinematicsCommandList;
+import us.ihmc.commonWalkingControlModules.controllerCore.command.lowLevel.LowLevelOneDoFJointDesiredDataHolder;
+import us.ihmc.commonWalkingControlModules.controllerCore.command.virtualModelControl.VirtualModelControlCommandList;
+
+public class ControllerCoreCommandDataHolder implements ControllerCoreCommandInterface
+{
+   private boolean hasInverseDynamicsCommandList = false;
+   private final InverseDynamicsCommandList inverseDynamicsCommandList = new InverseDynamicsCommandList();
+
+   private boolean hasInverseKinematicsCommandList = false;
+   private final InverseKinematicsCommandList inverseKinematicsCommandList = new InverseKinematicsCommandList();
+
+   private boolean hasVirtualModelControlCommandList = false;
+   private final VirtualModelControlCommandList virtualModelControlCommandList = new VirtualModelControlCommandList();
+
+   private boolean hasFeedbackControlCommandList = false;
+   private final FeedbackControlCommandList feedbackControlCommandList = new FeedbackControlCommandList();
+
+   private boolean hasLowLevelOneDoFJointDesiredDataHolder = false;
+   private final LowLevelOneDoFJointDesiredDataHolder lowLevelOneDoFJointDesiredDataHolder = new LowLevelOneDoFJointDesiredDataHolder();
+
+   private boolean hasControllerCoreMode = false;
+   private WholeBodyControllerCoreMode controllerCoreMode;
+
+   private boolean reinitialize = false;
+
+   public ControllerCoreCommandDataHolder()
+   {
+      clear();
+   }
+
+   public void clear()
+   {
+      hasControllerCoreMode = false;
+      hasFeedbackControlCommandList = false;
+      hasInverseDynamicsCommandList = false;
+      hasVirtualModelControlCommandList = false;
+      hasInverseKinematicsCommandList = false;
+      hasLowLevelOneDoFJointDesiredDataHolder = false;
+
+      inverseDynamicsCommandList.clear();
+      feedbackControlCommandList.clear();
+      inverseKinematicsCommandList.clear();
+      virtualModelControlCommandList.clear();
+      lowLevelOneDoFJointDesiredDataHolder.clear();
+      reinitialize = false;
+   }
+
+   public void set(ControllerCoreCommandDataHolder other)
+   {
+      hasInverseDynamicsCommandList = true;
+      hasInverseKinematicsCommandList = true;
+      hasVirtualModelControlCommandList = true;
+      hasControllerCoreMode = true;
+      hasFeedbackControlCommandList = true;
+      hasLowLevelOneDoFJointDesiredDataHolder = true;
```

Comment:
this should be getting the value from the other object, not setting them all to be true. For example, if other.getInverseDynamicsCommandList() is empty, then hasInverseDynamicsCommandList should be false.